src/librustc_trans/mir/block.rs

```diff
@@ -582,15 +577,15 @@ impl<'a, 'tcx> MirContext<'a, 'tcx> {
                     fn_ty.apply_attrs_callsite(invokeret);
 
                     if let Some((_, target)) = *destination {
-                        let ret_bcx = self.build_block(target);
-                        ret_bcx.at_start(|ret_bcx| {
-                            self.set_debug_loc(&ret_bcx, terminator.source_info);
-                            let op = OperandRef {
-                                val: Immediate(invokeret),
-                                ty: sig.output(),
-                            };
-                            self.store_return(&ret_bcx, ret_dest, fn_ty.ret, op);
-                        });
+                        let ret_bcx = self.get_builder(target);
+                        ret_bcx.position_at_start(ret_bcx.llbb());
+                        self.set_debug_loc(&ret_bcx, terminator.source_info);
+                        let op = OperandRef {
+                            val: Immediate(invokeret),
+                            ty: sig.output(),
+                        };
+                        self.store_return(&ret_bcx, ret_dest, fn_ty.ret, op);
+                        ret_bcx.position_at_end(ret_bcx.llbb());
```

Comment:
> The builder is dead after this point so it shouldn't be needed.

That’s not true if I remember this code correctly, but I’m confused why all the repositioning is necessary at all if the blocks are translated in reverse-postorder.

Can we see if the stuff works if you remove both repositions?